extensions/src/common/extension/decorators/newBlocks.ts

```diff
@@ -1,100 +1,29 @@
-
-            let textFunction;
-            if (typeof builderOrStrings == "function") {
-                // Get the tagged template string
-                const taggedTemplateLiteral = extractTaggedTemplateLiteral(builderOrStrings.toString());
-                const tagFunction = (strings1, ...values) => {
-                    // Set the arguments
-                    argList = values;
-                    // Set the text function
-                    textFunction = (...values: any[]) => {
-                        let result = '';
-                        strings1.forEach((str, index) => {
-                            result += str;
-                            if (index < values.length) {
-                                result += values[index];
-                            }
-                        });
-                        return result;
-                    };
-                    return "";
-
-                };
-                // Initiate the function?
-                tagFunction`For some reason, I need this line for the line below...`;
-                // Call the function and set arguments and text function
-                const taggedTemplate = eval("tagFunction`" + taggedTemplateLiteral + "`");
-
-            } else {
-                // Set the text function
-                textFunction = (...args: any[]) => {
-                    const strings = Array.isArray(builderOrStrings) ? builderOrStrings : [builderOrStrings];
-                    let result = '';
-                    strings.forEach((str, index) => {
-                        result += str;
-                        if (index < args.length) {
-                            result += args[index];
-                        }
-                    });
-                    return result;
-                };
-            }
-
-            // Push the block
-            //type Fn = (this: This extends ExtensionInstance, value: any, util: BlockUtilityWithID) => void;
-            type Fn = (...args) => Return;
-            const blockInfo = { type: blockType, text: textFunction, args: argList };
-            context.addInitializer(function () { this.pushBlock(opcode, blockInfo as BlockMetadata<Fn>, target) });
-            
-            // bundle events
-            const isProbableAtBundleTime = !isFunction(blockInfo);
-            if (isProbableAtBundleTime) {
-                const { type } = blockInfo;
-                blockBundleEvent?.fire({
-                    methodName: opcode,
-                    args: extractArgs(blockInfo as BlockMetadata<Fn>).map(a => isString(a) ? a : a.type),
-                    // is 'any' an issue? Likely!
-                    returns: type === "command" ? "void" : "any",
-                    scratchType: blockInfo.type
-                });
-            }
-
-            return (function () { return this[internalFuncName].call(this, ...arguments) });
+            const input: any = typeof builderOrStrings == "function"
```

Comment:
probably this shouldn't be `any` but I anticipate this being fairly challenging to make typescript happy about